Blockzilla/OverlayView.swift

```diff
@@ -198,3 +198,16 @@ extension OverlayView: KeyboardHelperDelegate {
     func keyboardHelper(_ keyboardHelper: KeyboardHelper, keyboardDidShowWithState state: KeyboardState) {}
     func keyboardHelper(_ keyboardHelper: KeyboardHelper, keyboardDidHideWithState state: KeyboardState) {}
 }
+
+extension UIPasteboard {
+    func urlAsync(callback: @escaping (URL?) -> Void) {
+        DispatchQueue.global().async {
+            let pastedItem = UIPasteboard.general.string
+
+            DispatchQueue.main.async {
```

Comment:
So I think it's a bit better design for this not to assume the callback wants to run on the main thread. I would let it call the callback on the global concurrent queue and then have the callback do what it wants, which in this case is run on the main thread as it does UI stuff. Thoughts @bkmunar and @boek ?

Reply:
@Sdaswani The more I think about it I think you’re right. Usually I’ve tried to keep thread hopping in the same context, but you have to start making a lot of assumptions as both the callee and the caller.